File: SWEA/D2/1288. 새로운 불면증 치료법/새로운 불면증 치료법.py
T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    n = int(input())
    numbers = []

    count = 1
    number = 1
    while len(numbers) < 10:

        # 숫자 n곱
        number = n * count
        count += 1

        # 숫자 세기
        for s in str(number):
            if s not in numbers:
                numbers.append(s)

    print(f"#{test_case} {number}")

File: SWEA/D2/1288. 새로운 불면증 치료법/새로운 불면증 치료법.java
import java.io.FileInputStream;
import java.util.HashSet;
import java.util.Scanner;
 
class Solution {
  public static void main(String args[]) throws Exception {
 
    Scanner sc = new Scanner(System.in);
    int T;
    T = sc.nextInt();
 
    for (int test_case = 1; test_case <= T; test_case++) {
      int number = sc.nextInt();
 
      HashSet<Integer> count = new HashSet<Integer>();
 
      int i = 1;
      int result = 0;
      while (count.size() < 10) {
        int ni = number * i;
        result = ni;
 
        for (String s : String.valueOf(ni).split("")) {
          count.add(Integer.valueOf(s));
        }
        i++;
      }
      System.out.println("#" + test_case + " " + result);
    }
  }
}